<a href="https://colab.research.google.com/github/soustan/Internship-Data-Analitic/blob/main/Internship_part_4.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

In [30]:
import pandas as pd
import numpy as np

In [ ]:
#Load the region roles
from google.colab import files
upload = files.upload()

In [40]:
reg_rol = pd.read_excel('region_roles_26_05_22.xlsx')

In [32]:
reg_rol.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 69656 entries, 0 to 69655
Data columns (total 34 columns):
 #   Column                    Non-Null Count  Dtype  
---  ------                    --------------  -----  
 0   Unnamed: 0                69656 non-null  int64  
 1   id                        69656 non-null  int64  
 2   premium                   69656 non-null  bool   
 3   name                      69656 non-null  object 
 4   department                5263 non-null   object 
 5   has_test                  69656 non-null  bool   
 6   response_letter_required  69656 non-null  bool   
 7   area                      69656 non-null  object 
 8   salary                    37590 non-null  object 
 9   type                      69656 non-null  object 
 10  address                   40318 non-null  object 
 11  response_url              1 non-null      object 
 12  sort_point_distance       0 non-null      float64
 13  published_at              69656 non-null  object 
 14  create

В ходе изучения Датасет, удалось оценить его размер, который равен 15.7 MB. Количество строк в датасет - 69656. Количество колонок(столбцов) - 34. В результате чего, стало понятным, что:
Количество булевых строк равно 5, Строк с плавающей запятой равно 1, строк с целым числом равно 2, строк с типом данных object равно 26. Столбцы, с именами id, premium, name, department, type, address и другие, имеют пустые или нулевые значения. В результате, из данных подлежащих анализу. имеются значения: Целые числа, значения отличные от нуля, а так же значения типа object, столбец salary.

In [38]:
reg_rol = reg_rol.astype'salary': np.int})

/usr/local/lib/python3.7/dist-packages/ipykernel_launcher.py:1: DeprecationWarning: `np.int` is a deprecated alias for the builtin `int`. To silence this warning, use `int` by itself. Doing this will not modify any behavior and is safe. When replacing `np.int`, you may wish to use e.g. `np.int64` or `np.int32` to specify the precision. If you wish to review your current use, check the release note link for additional information.
Deprecated in NumPy 1.20; for more details and guidance: https://numpy.org/devdocs/release/1.20.0-notes.html#deprecations
  """Entry point for launching an IPython kernel.


ValueError: ignored

In [2]:
#Load the result roles
from google.colab import files
upload = files.upload()

Saving result roles (16.06.2022).xlsx to result roles (16.06.2022).xlsx


In [23]:
res_rol = pd.read_excel('result_roles_16_06_2022.xlsx')

In [24]:
res_rol.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 76382 entries, 0 to 76381
Data columns (total 51 columns):
 #   Column                    Non-Null Count  Dtype 
---  ------                    --------------  ----- 
 0   Unnamed: 0                76382 non-null  int64 
 1   id                        76382 non-null  int64 
 2   premium                   76382 non-null  bool  
 3   name                      76382 non-null  object
 4   department                4902 non-null   object
 5   has_test                  76380 non-null  object
 6   response_letter_required  76382 non-null  object
 7   area                      76381 non-null  object
 8   salary                    40874 non-null  object
 9   type                      76379 non-null  object
 10  published_at              76380 non-null  object
 11  created_at                76381 non-null  object
 12  archived                  76380 non-null  object
 13  insider_interview         570 non-null    object
 14  url                   